warning in pio_darray.c

Here's a new warning:

```
pio_darray.c:268:102: warning: suggest parentheses around ‘&&’ within ‘||’ [-Wparentheses]
     if (iodesc->llen > 0 || (file->iotype==PIO_IOTYPE_NETCDF || file->iotype == PIO_IOTYPE_NETCDF4C) && ios->iomaster)
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~

```

I will fix and put up a PR.

--- a/src/clib/pio_darray.c
+++ b/src/clib/pio_darray.c
@@ -265,7 +265,9 @@ PIOc_write_darray_multi(int ncid, const int *varids, int ioid, int nvars,
      * large as the largest used to accommodate this serial io
      * method.  */
     rlen = 0;
-    if (iodesc->llen > 0 || (file->iotype==PIO_IOTYPE_NETCDF || file->iotype == PIO_IOTYPE_NETCDF4C) && ios->iomaster)
+    if (iodesc->llen > 0 ||
+        ((file->iotype == PIO_IOTYPE_NETCDF ||
+          file->iotype == PIO_IOTYPE_NETCDF4C) && ios->iomaster))
         rlen = iodesc->maxiobuflen * nvars;
 
     /* Allocate iobuf. */

